framework/src/main/java/org/checkerframework/framework/type/AnnotatedTypeFactory.java

```diff
@@ -3078,59 +3107,93 @@ public Set<AnnotationMirror> getWidenedAnnotations(
     }
 
     /**
-     * Returns the types of the two arguments to the BinaryTree, accounting for widening and
-     * unboxing if applicable.
+     * Returns the types of the two arguments to the BinaryTree. Please refer to {@link
+     * #binaryTreeArgTypes(TypeMirror, AnnotatedTypeMirror, AnnotatedTypeMirror)} )} for more
+     * details.
      *
      * @param node a binary tree
-     * @return the types of the two arguments
+     * @return the types of the two arguments (note that the underlying java types of both operands
+     *     will be converted to string if the binary operation is string concatenation)
      */
     public Pair<AnnotatedTypeMirror, AnnotatedTypeMirror> binaryTreeArgTypes(BinaryTree node) {
         return binaryTreeArgTypes(
-                getAnnotatedType(node.getLeftOperand()), getAnnotatedType(node.getRightOperand()));
+                TreeUtils.typeOf(node),
+                getAnnotatedType(node.getLeftOperand()),
+                getAnnotatedType(node.getRightOperand()));
     }
 
     /**
-     * Returns the types of the two arguments to the CompoundAssignmentTree, accounting for widening
-     * and unboxing if applicable.
+     * Returns the types of the two arguments to the CompoundAssignmentTree. Please refer to {@link
+     * #binaryTreeArgTypes(TypeMirror, AnnotatedTypeMirror, AnnotatedTypeMirror)} ) for more
+     * details.
      *
      * @param node a compound assignment tree
-     * @return the types of the two arguments
+     * @return the types of the two arguments (note that the underlying java types of the right
+     *     operand will be converted to string if the binary operation is string concatenation)
      */
     public Pair<AnnotatedTypeMirror, AnnotatedTypeMirror> compoundAssignmentTreeArgTypes(
             CompoundAssignmentTree node) {
         return binaryTreeArgTypes(
-                getAnnotatedType(node.getVariable()), getAnnotatedType(node.getExpression()));
+                TreeUtils.typeOf(node.getVariable()),
+                getAnnotatedType(node.getVariable()),
+                getAnnotatedType(node.getExpression()));
     }
 
     /**
-     * Returns the types of the two arguments to a binary operation, accounting for widening and
-     * unboxing if applicable.
+     * Returns the types of the two arguments to a binary operation. There are two special cases: 1.
```

Comment:
The `1.` starts at the end of the line. Maybe use markup to format as a nice list. Or add the line break before the number.

Reply:
Done